test/Microsoft.AspNetCore.Mvc.Razor.Test/RazorPageTest.cs

```diff
@@ -472,6 +472,23 @@ public void IsSectionDefined_ThrowsIfPreviousSectionWritersIsNotRegistered()
         }
 
         [Fact]
+        public async Task EnsureRenderedBodyOrSections_SucceedsIfRenderBodyIsNotCalledFromPage_AndNoSectionsAreDefined_AndBodyIgnored()
+        {
+            // Arrange
+            var path = "page-path";
+            var page = CreatePage(v =>
+            {
+            });
+            page.Path = path;
+            page.BodyContent = new HtmlString("some content");
+            page.IgnoreBody();
+
+            // Act & Assert (does not throw)
+            await page.ExecuteAsync();
+            page.EnsureRenderedBodyOrSections();
```

Comment:
The suggestion was to add a `v.WriteLiteral(expected);` inside `CreatePage` and do a `Assert.Equal(expected, page.RenderedContent);` at the end. The intent is to verify that it's suppresses `RenderBody` content and not any of the content written by the layout.